File: reference/SmartPharmacy-Kotlin/app/src/main/java/com/example/ui/screens/DistributorScreens.kt
package com.example.ui.screens

import android.content.Intent
import android.net.Uri
import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material.icons.outlined.Call
import androidx.compose.material.icons.outlined.Email
import androidx.compose.material.icons.outlined.LocationOn
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.model.Distributor
import com.example.ui.components.CustomButton
import com.example.ui.components.CustomSearchBar
import com.example.ui.components.CustomTextField
import com.example.ui.components.DistributorCard
import com.example.ui.theme.*
import com.example.ui.viewmodel.PharmacyViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DistributorListScreen(
    viewModel: PharmacyViewModel,
    onNavigateToAddDistributor: () -> Unit,
    onNavigateToDistributorDetails: (String) -> Unit
) {
    val distributors by viewModel.distributors.collectAsState()
    val searchQuery by viewModel.distributorSearch.collectAsState()

    val filtered = remember(distributors, searchQuery) {
        distributors.filter {
            it.name.contains(searchQuery, ignoreCase = true) ||
                    it.phone.contains(searchQuery, ignoreCase = true) ||
                    it.address.contains(searchQuery, ignoreCase = true)
        }
    }

    Scaffold(
        floatingActionButton = {
            FloatingActionButton(
                onClick = onNavigateToAddDistributor,
                containerColor = SecondaryTeal,
                contentColor = SurfaceWhite,
                shape = RoundedCornerShape(16.dp),
                modifier = Modifier.testTag("fab_add_distributor")
            ) {
                Row(
                    modifier = Modifier.padding(horizontal = 16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(imageVector = Icons.Default.Add, contentDescription = "Add Distributor")
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Add Distributor", fontWeight = FontWeight.Bold)
                }
            }
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(horizontal = 16.dp)
        ) {
            Spacer(modifier = Modifier.height(12.dp))

            CustomSearchBar(
                query = searchQuery,
                onQueryChange = { viewModel.setDistributorSearch(it) },
                placeholder = "Search 15+ distributors, phone, city..."
            )

            Spacer(modifier = Modifier.height(12.dp))

            Text(
                text = "Registered Distributors (${filtered.size})",
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                color = TextPrimary
            )

            Spacer(modifier = Modifier.height(8.dp))

            LazyColumn(
                verticalArrangement = Arrangement.spacedBy(10.dp),
                contentPadding = PaddingValues(bottom = 80.dp)
            ) {
                items(filtered, key = { it.id }) { dist ->
                    DistributorCard(
                        distributor = dist,
                        onClick = { onNavigateToDistributorDetails(dist.id) }
                    )
                }
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun AddDistributorScreen(
    viewModel: PharmacyViewModel,
    onNavigateBack: () -> Unit
) {
    val context = LocalContext.current

    var name by remember { mutableStateOf("") }
    var phone by remember { mutableStateOf("") }
    var address by remember { mutableStateOf("") }
    var gstNumber by remember { mutableStateOf("") }
    var email by remember { mutableStateOf("") }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Add Distributor", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = SurfaceWhite)
            )
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            CustomTextField(
                value = name,
                onValueChange = { name = it },
                label = "Distributor / Firm Name",
                placeholder = "e.g. Apex Pharma Wholesale",
                leadingIcon = Icons.Default.LocalShipping
            )

            CustomTextField(
                value = phone,
                onValueChange = { phone = it },
                label = "Phone Number",
                placeholder = "+91 98000 12345",
                leadingIcon = Icons.Default.Phone
            )

            CustomTextField(
                value = email,
                onValueChange = { email = it },
                label = "Email Address (Optional)",
                placeholder = "orders@distributor.com",
                leadingIcon = Icons.Default.Email
            )

            CustomTextField(
                value = address,
                onValueChange = { address = it },
                label = "Office / Warehouse Address",
                placeholder = "Street, Area, City",
                leadingIcon = Icons.Default.LocationOn,
                singleLine = false
            )

            CustomTextField(
                value = gstNumber,
                onValueChange = { gstNumber = it },
                label = "GST Number (Optional)",
                placeholder = "e.g. 27AAAAA0000A1Z5",
                leadingIcon = Icons.Default.Receipt
            )

            Spacer(modifier = Modifier.height(16.dp))

            CustomButton(
                text = "Save Distributor",
                containerColor = SecondaryTeal,
                onClick = {
                    if (name.isBlank() || phone.isBlank()) {
                        Toast.makeText(context, "Please enter name and phone number", Toast.LENGTH_SHORT).show()
                        return@CustomButton
                    }

                    val newDist = Distributor(
                        id = "D" + System.currentTimeMillis().toString().takeLast(4),
                        name = name,
                        phone = phone,
                        address = address.ifBlank { "Commercial Hub" },
                        gstNumber = gstNumber,
                        email = email.ifBlank { "info@$name.com" }
                    )

                    viewModel.addDistributor(newDist)
                    Toast.makeText(context, "Distributor Saved Successfully!", Toast.LENGTH_SHORT).show()
                    onNavigateBack()
                }
            )
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DistributorDetailsScreen(
    viewModel: PharmacyViewModel,
    distributorId: String,
    onNavigateBack: () -> Unit
) {
    val context = LocalContext.current
    val distributors by viewModel.distributors.collectAsState()
    val medicines by viewModel.medicines.collectAsState()

    val distributor = distributors.find { it.id == distributorId }

    if (distributor == null) {
        Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text("Distributor not found")
        }
        return
    }

    val suppliedMedicines = remember(distributor, medicines) {
        medicines.filter { it.distributorName.equals(distributor.name, ignoreCase = true) }
    }

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Distributor Profile", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = SurfaceWhite)
            )
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Distributor Header Card
            Card(
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = SecondaryTealLight),
                modifier = Modifier.fillMaxWidth()
            ) {
                Column(modifier = Modifier.padding(20.dp)) {
                    Text(text = distributor.name, fontSize = 20.sp, fontWeight = FontWeight.Bold, color = SecondaryTeal)
                    Spacer(modifier = Modifier.height(8.dp))

                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(imageVector = Icons.Outlined.Call, contentDescription = null, tint = SecondaryTeal, modifier = Modifier.size(18.dp))
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(text = distributor.phone, fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = TextPrimary)
                    }

                    Spacer(modifier = Modifier.height(4.dp))

                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(imageVector = Icons.Outlined.Email, contentDescription = null, tint = SecondaryTeal, modifier = Modifier.size(18.dp))
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(text = distributor.email, fontSize = 13.sp, color = TextSecondary)
                    }

                    Spacer(modifier = Modifier.height(4.dp))

                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(imageVector = Icons.Outlined.LocationOn, contentDescription = null, tint = SecondaryTeal, modifier = Modifier.size(18.dp))
                        Spacer(modifier = Modifier.width(6.dp))
                        Text(text = distributor.address, fontSize = 13.sp, color = TextSecondary)
                    }

                    if (distributor.gstNumber.isNotEmpty()) {
                        Spacer(modifier = Modifier.height(8.dp))
                        Surface(shape = RoundedCornerShape(8.dp), color = SurfaceWhite) {
                            Text(
                                text = "GSTIN: ${distributor.gstNumber}",
                                fontSize = 12.sp,
                                fontWeight = FontWeight.Bold,
                                color = SecondaryTeal,
                                modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp)
                            )
                        }
                    }
                }
            }

            // Quick Call / Mail Actions
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                Button(
                    onClick = {
                        val intent = Intent(Intent.ACTION_DIAL, Uri.parse("tel:${distributor.phone}"))
                        context.startActivity(intent)
                    },
                    colors = ButtonDefaults.buttonColors(containerColor = SecondaryTeal),
                    shape = RoundedCornerShape(12.dp),
                    modifier = Modifier
                        .weight(1f)
                        .height(46.dp)
                ) {
                    Icon(imageVector = Icons.Default.Call, contentDescription = null, modifier = Modifier.size(18.dp))
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Call Supplier")
                }

                OutlinedButton(
                    onClick = {
                        Toast.makeText(context, "Opening mail client for ${distributor.email}", Toast.LENGTH_SHORT).show()
                    },
                    shape = RoundedCornerShape(12.dp),
                    modifier = Modifier
                        .weight(1f)
                        .height(46.dp)
                ) {
                    Icon(imageVector = Icons.Default.Email, contentDescription = null, modifier = Modifier.size(18.dp))
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Email")
                }
            }

            // Supplied Medicines Section
            Text(
                text = "Supplied Medicines (${suppliedMedicines.size})",
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
                color = TextPrimary
            )

            if (suppliedMedicines.isEmpty()) {
                Text(
                    text = "No medicines associated with this distributor yet.",
                    fontSize = 13.sp,
                    color = TextSecondary
                )
            } else {
                Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                    suppliedMedicines.forEach { med ->
                        Card(
                            shape = RoundedCornerShape(12.dp),
                            colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
                            modifier = Modifier
                                .fillMaxWidth()
                                .border(1.dp, SurfaceBorder, RoundedCornerShape(12.dp))
                        ) {
                            Row(
                                modifier = Modifier
                                    .fillMaxWidth()
                                    .padding(12.dp),
                                horizontalArrangement = Arrangement.SpaceBetween,
                                verticalAlignment = Alignment.CenterVertically
                            ) {
                                Column {
                                    Text(text = med.name, fontSize = 14.sp, fontWeight = FontWeight.Bold, color = TextPrimary)
                                    Text(text = "Batch: ${med.batchNumber} • ${med.company}", fontSize = 12.sp, color = TextSecondary)
                                }
                                Text(text = "${med.quantity} in stock", fontSize = 13.sp, fontWeight = FontWeight.SemiBold, color = PrimaryBlue)
                            }
                        }
                    }
                }
            }
        }
    }
}

File: lib/screens/distributor_screens.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../core/theme/app_colors.dart';
import '../models/distributor.dart';
import '../providers/pharmacy_provider.dart';
import '../routes/app_routes.dart';
import '../widgets/common/custom_button.dart';
import '../widgets/common/custom_search_bar.dart';
import '../widgets/common/custom_text_field.dart';
import '../widgets/distributor/distributor_card.dart';

class DistributorListScreen extends StatefulWidget {
  const DistributorListScreen({super.key});

  @override
  State<DistributorListScreen> createState() => _DistributorListScreenState();
}

class _DistributorListScreenState extends State<DistributorListScreen> {
  late TextEditingController _searchController;

  @override
  void initState() {
    super.initState();
    final provider = context.read<PharmacyProvider>();
    _searchController = TextEditingController(text: provider.distributorSearch);
  }

  @override
  void dispose() {
    _searchController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final distributors = provider.distributors;
    final searchQuery = provider.distributorSearch;

    final filtered = distributors.where((d) {
      return d.name.toLowerCase().contains(searchQuery.toLowerCase()) ||
          d.phone.toLowerCase().contains(searchQuery.toLowerCase()) ||
          d.address.toLowerCase().contains(searchQuery.toLowerCase());
    }).toList();

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      floatingActionButton: FloatingActionButton.extended(
        onPressed: () => Navigator.pushNamed(context, AppRoutes.addDistributor),
        backgroundColor: AppColors.secondaryTeal,
        foregroundColor: AppColors.surfaceWhite,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
        icon: const Icon(Icons.add),
        label: const Text(
          "Add Distributor",
          style: TextStyle(fontWeight: FontWeight.bold),
        ),
      ),
      body: SafeArea(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: 12),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16.0),
              child: CustomSearchBar(
                controller: _searchController,
                hint: "Search 15+ distributors, phone, city...",
                onChanged: (val) => provider.setDistributorSearch(val),
              ),
            ),
            const SizedBox(height: 12),
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 16.0),
              child: Text(
                "Registered Distributors (${filtered.length})",
                style: const TextStyle(
                  fontSize: 14,
                  fontWeight: FontWeight.bold,
                  color: AppColors.textPrimary,
                ),
              ),
            ),
            const SizedBox(height: 8),
            Expanded(
              child: filtered.isEmpty
                  ? _buildEmptyState()
                  : ListView.separated(
                      padding: const EdgeInsets.fromLTRB(16, 0, 16, 80),
                      itemCount: filtered.length,
                      separatorBuilder: (context, index) => const SizedBox(height: 10),
                      itemBuilder: (context, index) {
                        final dist = filtered[index];
                        return DistributorCard(
                          distributor: dist,
                          onTap: () => Navigator.pushNamed(
                            context,
                            AppRoutes.distributorDetails,
                            arguments: dist.id,
                          ),
                        );
                      },
                    ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildEmptyState() {
    return Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: const [
          Icon(Icons.warning_amber_rounded, size: 48, color: AppColors.textMuted),
          SizedBox(height: 12),
          Text(
            "No distributors found",
            style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold, color: AppColors.textPrimary),
          ),
          Text(
            "Try adjusting your search query",
            style: TextStyle(fontSize: 13, color: AppColors.textSecondary),
          ),
        ],
      ),
    );
  }
}

class AddDistributorScreen extends StatefulWidget {
  const AddDistributorScreen({super.key});

  @override
  State<AddDistributorScreen> createState() => _AddDistributorScreenState();
}

class _AddDistributorScreenState extends State<AddDistributorScreen> {
  final _nameController = TextEditingController();
  final _phoneController = TextEditingController();
  final _addressController = TextEditingController();
  final _gstController = TextEditingController();
  final _emailController = TextEditingController();

  @override
  void dispose() {
    _nameController.dispose();
    _phoneController.dispose();
    _addressController.dispose();
    _gstController.dispose();
    _emailController.dispose();
    super.dispose();
  }

  void _saveDistributor() {
    final name = _nameController.text.trim();
    final phone = _phoneController.text.trim();
    final address = _addressController.text.trim();
    final gst = _gstController.text.trim();
    final email = _emailController.text.trim();

    if (name.isEmpty || phone.isEmpty) {
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(content: Text("Please enter name and phone number")),
      );
      return;
    }

    final newDist = Distributor(
      id: "D${DateTime.now().millisecondsSinceEpoch.toString().substring(DateTime.now().millisecondsSinceEpoch.toString().length - 4)}",
      name: name,
      phone: phone,
      address: address.isEmpty ? "Commercial Hub" : address,
      gstNumber: gst,
      email: email.isEmpty ? "info@${name.replaceAll(' ', '').toLowerCase()}.com" : email,
    );

    context.read<PharmacyProvider>().addDistributor(newDist);
    ScaffoldMessenger.of(context).showSnackBar(
      const SnackBar(content: Text("Distributor Saved Successfully!")),
    );
    Navigator.pop(context);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "Add Distributor",
          style: TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimary),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            CustomTextField(
              controller: _nameController,
              label: "Distributor / Firm Name",
              hint: "e.g. Apex Pharma Wholesale",
              leadingIcon: Icons.local_shipping,
            ),
            const SizedBox(height: 14),
            CustomTextField(
              controller: _phoneController,
              label: "Phone Number",
              hint: "+91 98000 12345",
              leadingIcon: Icons.phone,
              keyboardType: TextInputType.phone,
            ),
            const SizedBox(height: 14),
            CustomTextField(
              controller: _emailController,
              label: "Email Address (Optional)",
              hint: "orders@distributor.com",
              leadingIcon: Icons.email,
              keyboardType: TextInputType.emailAddress,
            ),
            const SizedBox(height: 14),
            CustomTextField(
              controller: _addressController,
              label: "Office / Warehouse Address",
              hint: "Street, Area, City",
              leadingIcon: Icons.location_on,
              singleLine: false,
            ),
            const SizedBox(height: 14),
            CustomTextField(
              controller: _gstController,
              label: "GST Number (Optional)",
              hint: "e.g. 27AAAAA0000A1Z5",
              leadingIcon: Icons.receipt,
            ),
            const SizedBox(height: 24),
            CustomButton(
              text: "Save Distributor",
              containerColor: AppColors.secondaryTeal,
              onPressed: _saveDistributor,
            ),
          ],
        ),
      ),
    );
  }
}

class DistributorDetailsScreen extends StatelessWidget {
  final String distributorId;

  const DistributorDetailsScreen({
    super.key,
    required this.distributorId,
  });

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final distributor = provider.distributors.firstWhere(
      (d) => d.id == distributorId,
      orElse: () => throw Exception("Distributor not found"),
    );

    final suppliedMedicines = provider.medicines.where((med) {
      return med.distributorName.toLowerCase() == distributor.name.toLowerCase();
    }).toList();

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "Distributor Profile",
          style: TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimary),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // Distributor Header Card
            Card(
              elevation: 0,
              margin: EdgeInsets.zero,
              shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
              color: AppColors.secondaryTealLight,
              child: Padding(
                padding: const EdgeInsets.all(20.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      distributor.name,
                      style: const TextStyle(
                        fontSize: 20,
                        fontWeight: FontWeight.bold,
                        color: AppColors.secondaryTeal,
                      ),
                    ),
                    const SizedBox(height: 8),
                    Row(
                      children: [
                        const Icon(Icons.call_outlined, color: AppColors.secondaryTeal, size: 18),
                        const SizedBox(width: 6),
                        Text(
                          distributor.phone,
                          style: const TextStyle(
                            fontSize: 14,
                            fontWeight: FontWeight.w600,
                            color: AppColors.textPrimary,
                          ),
                        ),
                      ],
                    ),
                    const SizedBox(height: 4),
                    Row(
                      children: [
                        const Icon(Icons.email_outlined, color: AppColors.secondaryTeal, size: 18),
                        const SizedBox(width: 6),
                        Text(
                          distributor.email,
                          style: const TextStyle(fontSize: 13, color: AppColors.textSecondary),
                        ),
                      ],
                    ),
                    const SizedBox(height: 4),
                    Row(
                      children: [
                        const Icon(Icons.location_on_outlined, color: AppColors.secondaryTeal, size: 18),
                        const SizedBox(width: 6),
                        Expanded(
                          child: Text(
                            distributor.address,
                            style: const TextStyle(fontSize: 13, color: AppColors.textSecondary),
                          ),
                        ),
                      ],
                    ),
                    if (distributor.gstNumber.isNotEmpty) ...[
                      const SizedBox(height: 8),
                      Container(
                        padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                        decoration: BoxDecoration(
                          color: Colors.white,
                          borderRadius: BorderRadius.circular(8),
                        ),
                        child: Text(
                          "GSTIN: ${distributor.gstNumber}",
                          style: const TextStyle(
                            fontSize: 12,
                            fontWeight: FontWeight.bold,
                            color: AppColors.secondaryTeal,
                          ),
                        ),
                      ),
                    ],
                  ],
                ),
              ),
            ),
            const SizedBox(height: 16),
            // Quick Call / Mail Actions
            Row(
              children: [
                Expanded(
                  child: ElevatedButton.icon(
                    onPressed: () {
                      // Simulating dialer behavior
                      ScaffoldMessenger.of(context).showSnackBar(
                        SnackBar(content: Text("Calling ${distributor.phone}...")),
                      );
                    },
                    icon: const Icon(Icons.call, size: 18),
                    label: const Text("Call Supplier"),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: AppColors.secondaryTeal,
                      foregroundColor: Colors.white,
                      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
                      fixedSize: const Size.fromHeight(46),
                    ),
                  ),
                ),
                const SizedBox(width: 10),
                Expanded(
                  child: OutlinedButton.icon(
                    onPressed: () {
                      ScaffoldMessenger.of(context).showSnackBar(
                        SnackBar(content: Text("Opening mail client for ${distributor.email}")),
                      );
                    },
                    icon: const Icon(Icons.email, size: 18),
                    label: const Text("Email"),
                    style: OutlinedButton.styleFrom(
                      foregroundColor: AppColors.textPrimary,
                      side: const BorderSide(color: AppColors.surfaceBorder),
                      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
                      fixedSize: const Size.fromHeight(46),
                    ),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 24),
            // Supplied Medicines Section
            Text(
              "Supplied Medicines (${suppliedMedicines.length})",
              style: const TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.bold,
                color: AppColors.textPrimary,
              ),
            ),
            const SizedBox(height: 12),
            if (suppliedMedicines.isEmpty)
              const Text(
                "No medicines associated with this distributor yet.",
                style: TextStyle(fontSize: 13, color: AppColors.textSecondary),
              )
            else
              Column(
                children: suppliedMedicines.map((med) {
                  return Container(
                    margin: const EdgeInsets.only(bottom: 8),
                    padding: const EdgeInsets.all(12),
                    decoration: BoxDecoration(
                      color: Colors.white,
                      borderRadius: BorderRadius.circular(12),
                      border: Border.all(color: AppColors.surfaceBorder),
                    ),
                    child: Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Expanded(
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: [
                              Text(
                                med.name,
                                style: const TextStyle(
                                  fontSize: 14,
                                  fontWeight: FontWeight.bold,
                                  color: AppColors.textPrimary,
                                ),
                              ),
                              Text(
                                "Batch: ${med.batchNumber} • ${med.company}",
                                style: const TextStyle(
                                  fontSize: 12,
                                  color: AppColors.textSecondary,
                                ),
                              ),
                            ],
                          ),
                        ),
                        Text(
                          "${med.quantity} in stock",
                          style: const TextStyle(
                            fontSize: 13,
                            fontWeight: FontWeight.w600,
                            color: AppColors.primaryBlue,
                          ),
                        ),
                      ],
                    ),
                  );
                }).toList(),
              ),
          ],
        ),
      ),
    );
  }
}
